src/main/java/com/github/manolo8/darkbot/gui/titlebar/ExtraButton.java

```diff
@@ -31,12 +32,44 @@
     private static final Set<ExtraMenuProvider> EXTRA_DECORATIONS = new LinkedHashSet<>();
     private static final Lazy<Void> clean = new Lazy.NoCache<>();
 
-    public static void setExtraDecorations(Stream<ExtraMenuProvider> provider) {
+    public static void setExtraDecorations(Stream<ExtraMenuProvider> provider, FeatureRegistry featureRegistry) {
         EXTRA_DECORATIONS.clear();
-        provider.forEach(EXTRA_DECORATIONS::add);
+        PluginExtraMenuProvider.plugins.clear();
+
+        provider.forEach(extra -> {
+            if (!(extra instanceof PluginExtraMenuProvider) && !(extra instanceof DefaultExtraMenuProvider)) {
+                Plugin plugin = featureRegistry.getFeatureDefinition(extra).getPlugin();
+                PluginExtraMenuProvider.plugins.computeIfAbsent(
+                        plugin,
+                        features -> new LinkedHashSet<>()).add(extra);
+            } else {
+                EXTRA_DECORATIONS.add(extra);
+            }
+        });
+
         clean.send(null);
     }
 
+    @Feature(name = "Plugin menu provider", description = "Provides plugin extra menus")
+    public static class PluginExtraMenuProvider implements ExtraMenuProvider {
+        private static final Map<Plugin, Set<ExtraMenuProvider>> plugins = new HashMap<>();
```

Comment:
```suggestion
        private static final Map<Plugin, Set<ExtraMenuProvider>> PLUGINS = new LinkedHashMap<>();
```
I'd make it capital case since this is a constant, also make it a linked map for iteration order